### DISCUSSION TITLE
Example for multi objective BO

### DISCUSSION THREAD

**Discussion by A-ep93** [meta-pytorch/botorch#993] [Q&A] [answered]
Hello all,
I am new to PyTorch and Botorch and I want to use multi-objective BO. I was wondering whether there is a simple example for multiobjective BO so that I can start from. I saw the tutorials on the website, but they use qEHVI and qParEGO. I think these are for parallelization and make those tutorials complicated. Even if calculations are serial, it is fine with me. Are there any other simpler examples of multi BO that I can use?

**Comment by saitcakmak** [comment, marked as answer]
Hi @A-ep93. Unfortunately, I don't know of a simpler BoTorch example for MOBO that I can point to ([Ax](https://ax.dev) may simplify things compared to BoTorch). The only part that makes those notebooks parallel is where they set `BATCH_SIZE = 4`. If you replace this with `BATCH_SIZE = 1`, then the acquisition functions (particularly the output of `optimize_acqf`) would recommend one point at a time. 

Looking a bit closer into the [notebook](https://botorch.org/tutorials/multi_objective_bo), `optimize_qparego_and_get_observation` constructs an `acq_func_list`, which you should not need if you're using `BATCH_SIZE = 1`, in which case this function would look very similar to `optimize_qehvi_and_get_observation`. Otherwise, the main BO loop, the part after "Perform Bayesian Optimization loop with ..." looks a bit long, though that is only because it is running three algorithms at once. The variables corresponding to each algorithm is conveniently labeled with `_random`, `_parego`, `_nehvi`; so running only a single one of them should be as simple as deleting the lines that involve the others :).

**Reply by A-ep93** [reply]
Thank you very much for your reply @saitcakmak.
